client/packages/invoices/src/InboundShipment/DetailView/Footer/Footer.tsx

```diff
@@ -66,31 +74,58 @@ interface FooterComponentProps {
 export const FooterComponent = ({ onReturnLines }: FooterComponentProps) => {
   const t = useTranslation();
   const { navigateUpOne } = useBreadcrumbs();
+  const { clearSelected } = useTableStore();
+  const { info } = useNotification();
+  const changeCampaignOrProgramModal = useEditModal();
 
   const { data } = useInbound.document.get();
-  const isManuallyCreated = !data?.linkedShipment?.id;
   const onDelete = useInbound.lines.deleteSelected();
   const onZeroQuantities = useInbound.lines.zeroQuantities();
   const selectedLines = useInbound.utils.selectedLines();
+  const { data: preference } = usePreference(PreferenceKey.UseCampaigns);
+  const { onChangeCampaignOrProgram } =
+    useInbound.lines.changeCampaignOrProgram(selectedLines);
+  const isDisabled = useIsInboundDisabled();
+  const isManuallyCreated = !data?.linkedShipment?.id;
+
+  const handleCampaignClick = () => {
+    if (isDisabled) {
+      info(
+        t('messages.cant-change-campaign-or-program-on-finalised-invoice')
+      )();
+    } else {
+      changeCampaignOrProgramModal.onOpen();
+    }
+  };
 
   const actions: Action[] = [
     {
       label: t('button.delete-lines'),
       icon: <DeleteIcon />,
       onClick: onDelete,
     },
-    {
-      label: t('button.return-lines'),
-      icon: <ArrowLeftIcon />,
-      onClick: () => onReturnLines(selectedLines),
-      shouldShrink: false,
-    },
+    ...(preference?.useCampaigns
+      ? [
+          {
+            label: t('button.change-campaign-or-program'),
+            icon: <ArrowRightIcon />,
+            onClick: () => handleCampaignClick,
```

Comment:
need to call 😢

Reply:
Err... no longer works for me if call added 👀 because it's creates an arrow function for handleClick reference but doesn't call...

Reply:
Opens for me 👀  adding the call doesn't open for me 👀